ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -89,45 +87,34 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
+      try {
+        ExecutorWithRetries.execute((Supplier<Future<Void>>) () -> {
           kafkaTopicClient.deleteTopics(
-              Collections.singletonList(dataSource.getKsqlTopic().getKafkaTopicName()));
-          return null;
-        }
-      }, "Could not delete the corresponding kafka topic: "
-           + dataSource.getKsqlTopic().getKafkaTopicName());
-
+              Collections.singletonList(
+                  dataSource.getKsqlTopic().getKafkaTopicName()));
+          return CompletableFuture.completedFuture(null);
+        });
+      } catch (final RuntimeException e) {
```

Comment:
`ExecutorWithRetries` job is to simplify the code around retrying.  As such, it would be good to avoid these catch blocks - i.e. can execute _not_ throw `Exception`?